velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -208,5 +214,42 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, dateSub) {
+  const auto dateSubInt32 = [&](std::optional<Date> date,
+                                std::optional<int32_t> value) {
+    return evaluateOnce<Date>("date_sub(c0, c1)", date, value);
+  };
+  const auto dateSubInt16 = [&](std::optional<Date> date,
+                                std::optional<int16_t> value) {
+    return evaluateOnce<Date>("date_sub(c0, c1)", date, value);
+  };
+  const auto dateSubInt8 = [&](std::optional<Date> date,
+                               std::optional<int8_t> value) {
+    return evaluateOnce<Date>("date_sub(c0, c1)", date, value);
+  };
+
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateSubInt32(std::nullopt, 1));
+  EXPECT_EQ(std::nullopt, dateSubInt16(std::nullopt, 1));
+  EXPECT_EQ(std::nullopt, dateSubInt8(std::nullopt, 1));
+
+  // Simple tests
```

Comment:
Make sure all comments are full sentences and end with a period.

Reply:
Done